### PR TITLE
[GEN-2110]: update RBAC permissions for UI

### PR DESCRIPTION
**Pull request by BenElferink**
This pull request includes significant updates to the role and cluster role definitions in the `cli/cmd/resources/ui.go` file. The changes primarily involve refining the permissions for various resources to better align with the required operations.

### Changes to Role and ClusterRole Definitions:

* [`cli/cmd/resources/ui.go`](diffhunk://#diff-c286e10d34710a80a59127b2b7951e8a33d9b9554e47d2f2b827fd690f2e53abL143-R156): Updated the `NewUIRole` function to refine the permissions for accessing `configmaps`, `secrets`, and `odigos.io` resources. This includes adding comments to clarify the necessity of each permission set.
* [`cli/cmd/resources/ui.go`](diffhunk://#diff-c286e10d34710a80a59127b2b7951e8a33d9b9554e47d2f2b827fd690f2e53abL231-R230): Updated the `NewUIClusterRole` function to refine the permissions for accessing `namespaces`, `services`, `pods`, `replicasets`, and `odigos.io` resources. Comments were added to explain the purpose of each permission set.

### CODE REVIEW

**Review by blumamir — COMMENTED** [review]
Thanks ben, really love the comments you've added to explain why each permission is used.

Also great that you merged few permissions into one to make it more compact and readable.

Left few more suggestions for additional improvement we can apply - to have minimal permissions, and ask for only what we need and use. this can help mitigate future bugs and looks better in security reviews.

The permissions modified here are for installing odigos with cli. [for helm, we maintain these permissions in another directory](https://github.com/odigos-io/odigos/tree/main/helm/odigos/templates/ui) which we currently need to sync manually. you can run `kubectl get role -n odigos-system odigos-ui -o yaml` and copy it (with minor changes to the metadata) to sync the helm template as well (do it also for clusterrole).

To test your changes, you can uninstall odigos and then run `make helm-install` from odigos root directory.

File: cli/cmd/resources/ui.go [hunk 1]
```diff
@@ -228,40 +194,40 @@ func NewUIClusterRole() *rbacv1.ClusterRole {
...
+			{ // Needed for "Describe Source", and for "Describe Odigos"
+				APIGroups: []string{"apps"},
+				Resources: []string{"replicasets"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{ // Needed to instrument applications
 				APIGroups: []string{"apps"},
-				Resources: []string{"deployments", "statefulsets", "daemonsets", "replicasets"},
+				Resources: []string{"deployments", "statefulsets", "daemonsets"},
 				Verbs:     []string{"get", "list", "watch", "patch", "update"},
```
Comment (blumamir) [inline cli/cmd/resources/ui.go:220]: one more thing we can clean up. the ui only watches for 3 CRDS [here](https://github.com/odigos-io/odigos/blob/3f17ebd0a20b5bbbc4e37bf17889b26e069f77e8/frontend/main.go#L321).

I think we can safely remove the "watch" permission from all other places. setting up a watcher is an expensive operation, and the permissions makes sure that if we add anything later on - it's intentional.

File: cli/cmd/resources/ui.go [hunk 2]
```diff
@@ -228,40 +194,40 @@ func NewUIClusterRole() *rbacv1.ClusterRole {
 			Name: "odigos-ui",
 		},
 		Rules: []rbacv1.PolicyRule{
-			{
+			{ // Needed to get and instrument namespaces
```
[the remark this answers is not in the document]
Comment (BenElferink) [reply cli/cmd/resources/ui.go:212]: Thank you 😊

**Comment by BenElferink** [comment]
Thanks @blumamir 
I made changes based on all the comments 😃

**Review by blumamir — APPROVED** [review]

**Pull request merged**